source/code/plugins/oms_common.rb

```diff
@@ -478,6 +480,141 @@ class Common
     @@tzRightFolder = 'right/'
 
     class << self
+
+      # Internal methods
+      # (left public for easy testing, though protected may be better later)
+
+      def clean_hostname_string(hnBuffer)
+        s = hnBuffer.strip
+        # Take off any port numbers:
+        if /\A(\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}):\d+\z/ =~ s
+            return $1
+        elsif not is_like_ipv6_string?(hnBuffer) and /\A(.*):\d+\z/ =~ s
+            return $1
+        end
+        return s
+      end
+
+      def has_Designated_HostnameFile?
+        return false if @@HostnameFilePath.nil?
+        return false if @@HostnameFilePath.length < 8 # 8 is first Guess.  Need policy here.
+        File.exist?(@@HostnameFilePath)
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return true if 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return false
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\.[12]?\d{1,2}\z/ =~ hnBuffer
+        qwa = hnBuffer.split('.')
+        return false unless qwa.length == 4
+        i = 0
+        qwa.each do |quadwordstring|
+            bi = quadwordstring.to_i
+            # This may need more detail if 255 octets are sometimes allowed, but I don't think so.
+            return false unless 0 <= bi and bi < 255
+            return false if i == 0 and bi == 0
+            i += 1
+        end
+        return false unless qwa[0].to_i != 0
+        return true
+      end
+
+      def is_like_ipv6_string?(hnBuffer)
+        return true if /\A\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}:\h{4}\z/ =~ hnBuffer
+        return false
```

Comment:
Why don't you do extra-thorough checks in this method like in `is_like_ipv4_string?()` ?

Reply:
Yes, the format I found for ipv6 was much simpler, because \h{4} is always fixed, and near as I can tell, since it has a 1 to 1 correspondence with the possible values already, there is no further need.  If I am missing something, please don't hesitate.  I only did some review of RFCs before I got started with these drafts, and could easily be making a mistake interpreting specfications.  Also, I did not look at all for any MS specifications that may further restrict our version of the protocol.